src/escape/escape.c

```diff
@@ -189,17 +211,46 @@ int main(int argc, char *argv[]) {
 
                         break;
 
-                case ACTION_UNESCAPE:
+                case ACTION_UNESCAPE: {
+                        _cleanup_free_ char *name = NULL;
+
+                        if (arg_template || arg_instance) {
+                                _cleanup_free_ char *template = NULL;
+
+                                r = unit_name_to_instance(*i, &name);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract instance: %m");
+                                        goto finish;
+                                }
+                                if (!*name) {
+                                        log_error("Unit %s is missing the instance name.", *i);
+                                        r = -EINVAL;
+                                        goto finish;
+                                }
+                                r = unit_name_template(*i, &template);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract template: %m");
+                                        goto finish;
+                                }
+                                if (arg_template && strcmp(arg_template, template)) {
```

Comment:
I think it is better to use `streq_ptr()`.

Reply:
Hm, I’ll change it to `streq` instead of `strcmp`, but IMHO it’s more readable to keep the `arg_template` nullness check separate instead of using `streq_ptr` – it’s not like this is some pointer that just may be null as well, its presence or absence is a flag which decides if we want to do this check at all or not.

(I guess that could also be expressed as an explicit `!arg_instance`? And then use `streq_ptr` again, but this time just to guard against the case of the conflict detection between `--instance` and `--template` being buggy.)